sdk/test-utils/multi-version/src/multiVersion.ts

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { isLiveMode } from "@azure/test-utils-recorder";
+import { getGlobalObject } from "./global";
+
+export interface TestFunctionWrapper {
+  it:
+    | Mocha.TestFunction
+    | (Mocha.PendingTestFunction & {
+        only: Mocha.ExclusiveTestFunction;
+      });
+  xit: Mocha.PendingTestFunction;
+  describe:
+    | Mocha.SuiteFunction
+    | (Mocha.PendingSuiteFunction & {
+        only: Mocha.ExclusiveSuiteFunction;
+      });
+  xdescribe: Mocha.PendingSuiteFunction;
+  global: NodeJS.Global;
+}
+
+export function lessThanOrEqual(
+  a: string,
+  b: string,
+  compareFunc?: (a: string, b: string) => number
+): boolean {
+  if (compareFunc) {
+    return compareFunc(a, b) <= 0;
+  }
+  return a <= b;
```

Comment:
I don't think this will work for semver generically, since:

![image](https://user-images.githubusercontent.com/639216/100684882-13e2a780-3330-11eb-97d4-b7ee7a7c7708.png)

Reply:
Not all services use semver for their api versions.  Storage has versions like "2019-07-07", "2019-12-12", "2020-02-10", etc. So I originally thought that a custom compare function should be passed in if string comparison doesn't work.  Maybe I am making it over-complicate.  We could just take an array and assume versions are sorted already. It's is then up to the test author to ensure the order is correct.

.NET defines their versions in the product as integer-backed enums and keeps a mapping between the enums and version strings.  I don't know whether we want to go that route.

Reply:
I'm fine forcing the versions to be sorted on input for now if that simplifies things